modules/api/src/main/java/org/apache/ignite/lang/util/IgniteNameUtils.java

```diff
@@ -213,29 +222,44 @@ public boolean hasNext() {
                         // looks like we just found a closing quote
                         sb.append(source, start, currentPosition);
 
+                        foundDot = hasNextChar();
                         currentPosition += 2;
 
                         return sb.toString();
                     }
 
-                    throwMalformedNameException();
-                } else if (!quoted && (currentChar() == '.' || currentChar() == ' ')) {
+                    throwMalformedIdentifierException();
+                } else if (c == '.') {
+                    if (quoted) {
+                        continue;
+                    }
+
                     sb.append(source, start, currentPosition);
 
                     currentPosition++;
+                    foundDot = true;
 
                     return sb.toString().toUpperCase();
+                } else if (!quoted
+                        && !identifierStart(source.codePointAt(currentPosition))
+                        && !identifierExtend(source.codePointAt(currentPosition))
```

Comment:
Including alphabetic characters (`indentifierStart`) into `identifierExtend` will make this code simpler.

Reply:
No, it doesn't.